### PR TITLE
19361 - Add refund status to routing slip

### PR DESCRIPTION
**Pull request by Jxio**
*Issue #:*
https://github.com/bcgov/entity/issues/19361

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

File: pay-api/migrations/versions/2024_10_22_a430eb2b744e_add_refund_status_to_routing_slip.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+def upgrade():
+    op.execute("set statement_timeout=20000;")
+    with op.batch_alter_table('routing_slips', schema=None) as batch_op:
+        batch_op.add_column(sa.Column('refund_status', sa.String(length=50), nullable=True))
+
+    op.execute("""
+        UPDATE routing_slip_status_codes
+        SET code = 'REFUND_PROCESSED', description = 'Refund Processed'
+        WHERE code = 'REFUND_COMPLETED';
```
Comment (seeker25) [inline pay-api/migrations/versions/2024_10_22_a430eb2b744e_add_refund_status_to_routing_slip.py:34]: Use the enum please

File: pay-api/migrations/versions/2024_10_22_a430eb2b744e_add_refund_status_to_routing_slip.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    # Update the records in routing_slips where status is 'REFUND_COMPLETED'
+    op.execute("""
+        UPDATE routing_slips
+        SET status = 'REFUND_PROCESSED'
+        WHERE status = 'REFUND_COMPLETED';
+    """)
+
+
+def downgrade():
+    op.execute("set statement_timeout=20000;")
```
Comment (seeker25) [inline pay-api/migrations/versions/2024_10_22_a430eb2b744e_add_refund_status_to_routing_slip.py:46]: low traffic table not needed

**Comment by seeker25** [comment]
You're still missing pay-queue changes

image

File: pay-api/src/pay_api/services/refund.py [hunk 3]
```diff
@@ -219,6 +219,9 @@ def create_routing_slip_refund(cls, routing_slip_number: str, request: Dict[str,
             reason = f"Refund Rejected by {user_name}"
 
         rs_model.status = refund_status
+        rs_model.refund_status = (RoutingSlipRefundStatus.PROCESSED
```
Comment (seeker25) [inline pay-api/src/pay_api/services/refund.py:222]: I think this should be set to processed in pay-queue when the makes it to `RoutingSlipStatus.REFUND_PROCESSED`

Comment (Jxio) [reply pay-api/src/pay_api/services/refund.py:222]: I think it should not set PROCESSED directly here, just looking at refund fas flow, it doesn't make sense refund directly without any feedback file received. Need to double check.

File: jobs/payment-jobs/poetry.lock [hunk 4]
```diff
@@ -2,98 +2,113 @@
 
 [[package]]
 name = "aiohappyeyeballs"
```
Comment (seeker25) [inline jobs/payment-jobs/poetry.lock:4]: Please revert this, just do `poetry update pay-api`

File: pay-api/src/pay_api/models/routing_slip.py [hunk 5]
```diff
@@ -70,6 +70,7 @@ class RoutingSlip(Audit):  # pylint: disable=too-many-instance-attributes
             "updated_by",
             "updated_name",
             "updated_on",
+            "refund_status"
```
Comment (seeker25) [inline pay-api/src/pay_api/models/routing_slip.py:73]: `NOTE: please keep mapper names in alpha-order, easier to track that way`

File: pay-api/src/pay_api/services/fas/routing_slip_status_transition_service.py [hunk 6]
```diff
@@ -55,7 +55,7 @@ class RoutingSlipStatusTransitionService:  # pylint: disable=too-many-instance-a
             RoutingSlipCustomStatus.CANCEL_WRITE_OFF_REQUEST.custom_status,  # pylint: disable=no-member
         ],
         RoutingSlipStatus.REFUND_AUTHORIZED.value: [],
-        RoutingSlipStatus.REFUND_COMPLETED.value: [],
+        RoutingSlipStatus.REFUND_PROCESSED.value: [],
```
Comment (seeker25) [inline pay-api/src/pay_api/services/fas/routing_slip_status_transition_service.py:58]: Have you handled the changes in FAS-UI for this?

File: pay-api/src/pay_api/services/refund.py [hunk 7]
```diff
@@ -216,7 +216,10 @@ def create_routing_slip_refund(cls, routing_slip_number: str, request: Dict[str,
         # Rejected refund makes routing slip active
         if refund_status == RoutingSlipStatus.REFUND_REJECTED.value:
```
Comment (seeker25) [inline pay-api/src/pay_api/services/refund.py:217]: rename this to `status == RoutingSlipStatus.REFUND_REJECTED.value`

so it's not confusing with the true `refund_status`

File: pay-queue/poetry.lock [hunk 8]
```diff
@@ -1,3189 +1,3512 @@
-# This file is automatically @generated by Poetry 1.8.3 and should not be changed by hand.
-
-[[package]]
-name = "aiohappyeyeballs"
```
Comment (seeker25) [inline pay-queue/poetry.lock:4]: Revert this please, unless you plan on testing it  - `poetry update pay-api`

File: pay-api/src/pay_api/services/fas/routing_slip_status_transition_service.py (hunk 6 above)
Comment (Jxio) [reply pay-api/src/pay_api/services/fas/routing_slip_status_transition_service.py:58]: yes

**Review by seeker25 — APPROVED** [review]

**Pull request merged**